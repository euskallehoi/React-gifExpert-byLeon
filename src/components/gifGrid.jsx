import { useState, useEffect } from 'react';

import { getGifs } from '../helpers/getgifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './gifGridItem';


export const GifGrid = ({ category }) => {
  
    const { images, isLoading } = useFetchGifs( category )



    
    return (
        <>
            <h3>{ category }</h3>
            {
                isLoading  &&  <h2>cargando...</h2>
            }

            <div className='card-grid'>
                {
                    images.map( (image) => (
                        <GifItem 
                            key={ image.id }
                            {...image}
                        />
                    ))
                }
            </div>
        </>
    )
}