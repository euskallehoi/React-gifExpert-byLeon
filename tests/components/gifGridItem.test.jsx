import { GifItem } from '../../src/components/gifGridItem'
import {render, screen} from '@testing-library/react'

describe('Pruebas en <gifItem />', () => {

    const title = 'Piccolo';
    const url = 'https://dragon-ball.com/piccolo.jpn';

    test('debe de hacer match con el snapchot', () => {
        
        const { container } = render( <GifItem title={ title } url={ url } /> )
        expect( screen.getByText( title ) ).toBeTruthy();

    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        
        render( <GifItem title={ title } url={ url } /> )
        //screen.debug();
        // expect(screen.getByRole('img').src ).toBe( url );
        // expect(screen.getByRole('img').src ).toBe( title );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    });

    test('debe de mostrar el titulo en el componente', () => {
        
        render( <GifItem title={ title } url={ url } /> )
        expect( screen.getByText( title ) ).toBeTruthy();

    });
})
