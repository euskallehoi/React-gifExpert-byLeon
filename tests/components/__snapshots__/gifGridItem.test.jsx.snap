// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <gifItem /> debe de hacer match con el snapchot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Piccolo"
      src="https://dragon-ball.com/piccolo.jpn"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
